android/src/main/java/com/swmansion/rnexecutorch/utils/Fetcher.kt

```diff
@@ -132,85 +124,132 @@ class Fetcher {
       return response
     }
 
+    private fun getIdOfResource(context: Context, resourceName: String): Int {
+      return context.resources.getIdentifier(resourceName, "raw", context.packageName)
+    }
+
     fun downloadResource(
       context: Context,
       client: OkHttpClient,
-      url: URL,
+      url: String,
       resourceType: ResourceType,
+      isLargeFile: Boolean,
       onComplete: (String?, Exception?) -> Unit,
       listener: ProgressResponseBody.ProgressListener? = null,
     ) {
       /*
       Fetching model and tokenizer file
-      1. Extract file name from provided URL
-      2. If file name contains / it means that the file is local and we should return the path
-      3. Check if the file has a valid extension
-          a. For tokenizer, the extension should be .bin
-          b. For model, the extension should be .pte
-      4. Check if models directory exists, if not create it
-      5. Check if the file already exists in the models directory, if yes return the path
-      6. If the file does not exist, and is a tokenizer, fetch the file
-      7. If the file is a model, fetch the file with ProgressResponseBody
+      1. Check if the provided file is a bundled local file
+         a. Check if it exists
+         b. Check if it has valid extension
+         c. Copy the file and return the path
+      2. Check if the provided file is a path to a local file
+         a. Check if it exists
+         b. Check if it has valid extension
+         c. Return the path
+      3. The provided file is a remote file
+          a. Check if it has valid extension
+          b. Check if it's a large file
+              i. Create temporary file to store it at download time
+              ii. Move it to the models directory and return the path
+          c. If it's not a large file download it and return the path
        */
-      val fileName: String
 
       try {
-        fileName = extractFileName(url)
-      } catch (e: Exception) {
-        onComplete(null, e)
-        return
-      }
+        if (!url.contains("://")) {
+          //The provided file is from react-native assets folder in release mode
+          val resId = getIdOfResource(context, url)
+          val resName = context.resources.getResourceEntryName(resId)
+          val fileExtension = if (resourceType == ResourceType.TOKENIZER) "bin" else "pte"
```

Comment:
maybe we can move this to a function, i think it is likely that we will re-use this, also this might be a problem if we add any other resource type